### DISCUSSION TITLE
[2024-11-25] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#11651] [Weekly Demo Day]

**Comment by farmaazon** [comment]
* Finished #11547

https://github.com/user-attachments/assets/a528e26a-b388-4a2a-9bf4-3ccc734373f6

* [Work in improving our E2E tests](#11595): added traces uploading on CI, so debugging should be simpler, fixed one problem, the rest seem to be gone after fixing many network requests sent by Project's list.

**Comment by vitvakatu** [comment]
- Currently working on #11350

A bunch of small tasks:
- #11541 – slightly changing behavior when editing text and number literals. Now, defocusing input accepts the edit, and only <kbd>Esc</kbd> cancels.
- #11523 – the `Write` menu item allows to execute the node once in the `Live` context
- #11578 – Selected nodes now change colors, porting changes from James’ implementation.
- #11582 – Partially transparent visualization is tricky to implement without a significant performance drop; we choose to have imperfect edge rendering under visualization until we have time to fix it properly.
- #11590 – Self-explanatory, porting changes from James’ PoC.
- #11554 – Mitigating Chrome bug that causes focus issues in the code editor.
- #11592 – Minor changes to items in the component menu.
- #11612 – Using new API for creating table literals.


https://github.com/user-attachments/assets/ae9693be-d5ca-42cc-897c-761f311fcb91

**Comment by kazcw** [comment]
https://github.com/user-attachments/assets/b9193b5a-8db9-416e-a357-389532c170d9

- #11564
  - Also refactor CodeEditor, yCollab plugin

- #11585: Eliminates last Lexical dependency
  - Added integration tests for component comments

- #11597: Rendering, inserting when pasting

https://github.com/user-attachments/assets/d8c039e6-670c-4ff5-9d71-c07ee6114570

- #11620 

- #11621

**Comment by hubertp** [comment]
Relevant backend work:
- Healthcheck for polyglot Ydoc #11545
- Rename fix for unqualified method calls #11556 
- Intersection types support in LS #11583

Work on suspend - broken to some extent on Ydoc NodeJS and completely on polyglot Ydoc.

**Comment by Frizi** [comment]
- First step of function signature editor almost complete, finishing review fixes - https://github.com/enso-org/enso/pull/11655
<img width="439" alt="image" src="https://github.com/user-attachments/assets/73e6bb92-235f-497d-9cff-126dc4110f8b">
<img width="263" alt="image" src="https://github.com/user-attachments/assets/cb61f62a-755c-4865-ba6c-ab9130167713">

- Fixed issues with numeric widgets - https://github.com/enso-org/enso/issues/11557
- Fixed minor visual bugs around node visualizations
- Fixed environment variables setup in CI that affected displayed version and ag-grid token - https://github.com/enso-org/enso/pull/11584


![image](https://github.com/user-attachments/assets/6a0e9260-4eb1-4a4d-a215-7550810b185d)
